packages/domain-picker/src/constants.ts

```diff
@@ -11,6 +16,10 @@ export const PAID_DOMAINS_TO_SHOW_EXPANDED = 10;
  */
 export const DOMAIN_SEARCH_DEBOUNCE_INTERVAL = 300;
 
-export const DOMAIN_SUGGESTIONS_STORE = 'automattic/domains/suggestions';
+// TODO: Check domain suggestions store registration discrepancies
+// (see https://github.com/Automattic/wp-calypso/issues/46869)
+export const DOMAIN_SUGGESTIONS_STORE = DomainSuggestions.register( {
+	vendor: 'variation2_front',
+} );
```

Comment:
I also updated #46869 accordingly

Reply:
Hmm, after doing this here, now I realized why the domain store was being registered on the parent (e.g. gutenboarding). It is because gutenboarding, sitting inside `client/landing` can access calypso lib and config and register the store using the param values extracted from calypso lib and config.

Now I see why this is tricky.

Reply:
Should we then just call `register` without any parameters?

I assume that, when used inside Gutenboarding, the parent would register the store first (using the correct parameters from the config).
Then the domain picker would also attempt to register, but because of the singleton logic, the store won't be registered again and only the store key will be returned.

We'd just need to confirm this theory by checking the order in which the `register` function gets called

Reply:
By checking the logic of [getSuggestionsVendor](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/suggestions/index.js) and [how it's being used in Calypso signup flow](https://github.com/Automattic/wp-calypso/blob/master/client/signup/steps/domains/index.jsx#L522) (which is also handling site launch), I think we should use `'variation4_front'` for step-by-step and focused launch, same as in Gutenboarding. 

cc @daledupreez @hambai to confirm.

Also, when hardcoding this, please add a comment to mention `getSuggestionsVendor` that should've been used to get the correct value.

Reply:
The short version here is that the correct vendor _really_ depends on which flow you're in, and I don't know which one this package will be used in without digging in much deeper.

As @razvanpapadopol mentions, for signup flows, we want `variation4_front`, while for launch flows (which we assume is post-signup), we would want `variation2_front`.

After looking at the code here, we should also be clear about whether we want to show premium domains in this launch flow. For now, I don't think we should, but we should make sure we discuss that more generally, as we may want to start showing premium domains to users outside of the signup flow who happen to be using this flow. (There is a design/implementation cost to handling premiums, so we should budget for it before we decide to add them.)

Reply:
Also, might it make sense to externalize the `getSuggestionsVendor()` logic so it can be accessed more cleanly without a major dependency within Calypso. (If we did that, we would probably want to refactor the current use of `config` to have something more context-driven.)

Reply:
> for signup flows, we want variation4_front, while for launch flows (which we assume is post-signup), we would want variation2_front.

Then I think we have a bug currently in Calypso launch flow coming from [this line](https://github.com/Automattic/wp-calypso/blob/master/client/signup/steps/domains/index.jsx#L522) where `isSignup` option should have the value `'launch-site' !== this.props.flowName` instead of `true`.

**To reproduce:**
- Press _Launch your site_ tab and then _Launch site_ button in _Site setup_ checklist in My Home for an unlaunched site.
- You should land on `https://wordpress.com/start/launch-site?siteSlug=[SITE_SLUG]`.
- `variation4_front` is used to get domain suggestions, same as during signup flow.

Reply:
Should we merge this PR as-is, and move the conversation to one or more new issues?

The topics to discuss / actions to take seem to be:
- The bug mentioned by @razvanpapadopol in the [comment above](https://github.com/Automattic/wp-calypso/pull/46911#discussion_r516671976)
- Refactor and externalize the [`getSuggestionsVendor()`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/suggestions/index.js#L15-L23) function, 
- Refactor the `@automattic/domain-picker` and `@automattic/plans-grid` packages, since they are used both during signup and during site launch. They will need to somehow allow their parent module (i.e. `@automattic/launch` and [Gutenboarding](https://github.com/Automattic/wp-calypso/blob/master/client/landing/gutenboarding/stores/domain-suggestions/index.ts)) to set the correct vendor

Reply:
Apologies for the delay, but I think pushing those additional work items to separate PRs is the way to go here.

Reply:
I went ahead and opened these 3 issues:
- #47099
- #47101
- #47105 

Please feel free to edit/comment on those issues if you feel like I missed something.

I'm going to proceed and merge this PR as-is.